delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/indexing/IndexingStreamBehaviour.scala

```diff
@@ -0,0 +1,284 @@
+                  current match {
+                    case None                                                             =>
+                      UIO.delay(
+                        logger
+                          .debug(
+                            "Index view {} with revision {} in project {} will start from offset {}",
+                            id,
+                            fetchView.rev,
+                            project,
+                            progress.offset
+                          )
+                      ) >>
+                        StreamSwitch.run(
+                          streamName(fetchView.rev),
+                          buildStream(fetchView, progress),
+                          retryStrategy,
+                          onFinalize,
+                          onFinalize
+                        )
+                    case Some((currentIndex, switch)) if fetchView.rev > currentIndex.rev =>
+                      UIO.delay(
+                        logger.debug(
+                          "New revision of view {} in project {} is detected, the indexing will be restart from the beginning",
+                          fetchView.rev,
+                          id,
+                          project
+                        )
+                      ) >>
+                        switch.stop >> clearIndex(currentIndex.index) >> restartStream(fetchView)
+                    case Some((_, switch))                                                =>
+                      UIO.delay(
+                        logger.debug(
+                          " Same revision {} of view {} in project {}, we just carry on with the ongoing stream",
+                          fetchView.rev,
+                          id,
+                          project
+                        )
+                      ) >>
+                        UIO.pure(switch)
+                  }
+              } yield Started(fetchView, switch)
+          }
+
+          context.pipeToSelf(io.runToFuture) {
+            case Success(value) => value
+            case Failure(cause) => IndexingError(cause)
+          }
+        }
+
+        // Starts a new stream from the beginning
+        def restartStream(
+            viewIndex: ViewIndex[V]
+        ): Task[StreamSwitch] =
+          StreamSwitch.run(
+            streamName(viewIndex.rev),
+            buildStream(viewIndex, ProjectionProgress.NoProgress),
+            retryStrategy,
+            onFinalize,
+            onFinalize
+          )
+
+        // Inits the indexing
+        def init(): Behavior[IndexingViewCommand[V]] = {
+          logger.debug("Init indexing for view {} in project", id, project)
+          runStream(None)
+          balancing()
+        }
+
+        // State after init or after reset and when view gets updated
+        def balancing(): Behavior[IndexingViewCommand[V]] = Behaviors
+          .receiveMessage[IndexingViewCommand[V]] {
+            case Started(v, s)    =>
```

Comment:
don't you need any `ClassTag` on the pattern match here? How does it know the type after erasure?

Reply:
It comes from the way, the command ADT is written